provisioning/device/tests/RetryPolicies/ProvisioningClientExponentialBackoffRetryPolicyTests.cs

```diff
@@ -50,5 +50,18 @@ public void ExponentialBackoffRetryPolicy_IsExponential(uint retryCount)
             // assert
             delay.TotalMilliseconds.Should().BeApproximately(Math.Pow(2, exponent), 100);
         }
+
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public void ExponentialBackoffRetryPolicy_ProvisioningClientException_ReturnsTrueWhenTransient(bool isTransient)
+        {
+            // arrange
+            var retryPolicy = new ProvisioningClientExponentialBackoffRetryPolicy(0, TimeSpan.FromMinutes(100), false);
+            var ex = new ProvisioningClientException("", isTransient);
+
+            // act - assert
+            retryPolicy.ShouldRetry(1, ex, out _).Should().Be(isTransient);
```

Comment:
Since these retry policy tests would be needed for all clients, maybe someone could take the lead and make sure we're covering all scenarios for all of our clients.

Reply:
Yeah, sure, I would love to take care of that!